### PR TITLE
Delete global namespace

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Add support to delete global namespace.

**Why?**
Some checks needs to be added to support deletion of global namespace.

**How did you test it?**
Manually in local "global" cluster.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: service/history/workflow/mutable_state_impl.go [hunk 1]
```diff
@@ -4141,7 +4141,9 @@ func (e *MutableStateImpl) updateWithLastWriteEvent(
 }
 
 func (e *MutableStateImpl) canReplicateEvents() bool {
-	return e.namespaceEntry.ReplicationPolicy() == namespace.ReplicationPolicyMultiCluster
+	return e.namespaceEntry.ReplicationPolicy() == namespace.ReplicationPolicyMultiCluster &&
+		// Don't replicate workflow executions in deleted namespace.
+		e.namespaceEntry.State() != enumspb.NAMESPACE_STATE_DELETED
```
Comment (yycptt) [inline service/history/workflow/mutable_state_impl.go:4146]: cc @yux0

File: service/history/transferQueueStandbyTaskExecutor.go [hunk 2]
```diff
@@ -620,19 +627,20 @@ func (t *transferQueueStandbyTaskExecutor) fetchHistoryFromRemote(
...
 			0,
 		)
-	} else {
-		err = serviceerror.NewInternal(
-			"transferQueueStandbyProcessor encountered empty historyResendInfo",
-		)
 	}
 
 	if err != nil {
+		if _, isNotFound := err.(*serviceerror.NamespaceNotFound); isNotFound {
```
Comment (yycptt) [inline service/history/transferQueueStandbyTaskExecutor.go:633]: nit: can we define a util function for this check? maybe in the `common` package?

**Review by yycptt — APPROVED** [review]
The task processing part LGTM. @yux0 Would you mind go over the replication part?

File: service/history/transferQueueStandbyTaskExecutor.go (hunk 2 above)
Comment (alexshtin) [reply service/history/transferQueueStandbyTaskExecutor.go:653]: no. I am against this. go codders should suffer from all go features!

**Pull request merged**